### ISSUE TITLE
pip install -e '.[soap] fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8642]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8642|
|Type|enhancement|
|Created|2016-07-21 00:45:01Z|

If I do this on Python 3:

```
pip install -e ".[soap]"
```

It fails with:

```
Requirement already satisfied (use --upgrade to upgrade): zope.interface>=4.0.2 in /Users/crodrigues/venv/lib/python3.6/site-packages (from Twisted==16.3.0)
Collecting soappy (from Twisted==16.3.0)
  Using cached SOAPpy-0.12.22.zip
Requirement already satisfied (use --upgrade to upgrade): setuptools in /Users/crodrigues/venv/lib/python3.6/site-packages (from zope.interface>=4.0.2->Twisted==16.3.0)
Collecting wstools (from soappy->Twisted==16.3.0)
  Using cached wstools-0.4.3.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-build-dghs28oh/wstools/setup.py", line 19, in <module>
        from src.wstools.version import __version__
      File "/private/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-build-dghs28oh/wstools/src/wstools/__init__.py", line 6, in <module>
        import WSDLTools
    ImportError: No module named 'WSDLTools'
    
    ----------------------------------------
```

The released version of WSDLTools 0.4.3 does not work on Python 3: https://pypi.python.org/pypi/wstools/

WSDLTools in github ( https://github.com/pycontribs/wstools ) claims to support Python 3, but it isn't released yet.

<details><summary>Searchable metadata</summary>

```
trac-id__8642 8642
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469061901134192 1469061901134192
changetime__1469394330735239 1469394330735239
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc set status to `closed`|
|-|-|

I sent an e-mail to the maintainer of wstools,
and he released wstools 0.4.4 on July 22, 2016 and uploaded it to pypi: https://pypi.io/project/wstools/

Now if I use Python 3, and type 

```
pip install -e '.[soap]'
```

It works:

```
 pip install -e '.[soap]'
Obtaining file:///Users/crodrigues/twisted4
Collecting zope.interface>=4.0.2 (from Twisted==16.3.0)
Collecting soappy (from Twisted==16.3.0)
  Using cached SOAPpy-0.12.22.zip
Requirement already satisfied (use --upgrade to upgrade): setuptools in /Users/crodrigues/venv2/lib/python3.6/site-packages (from zope.interface>=4.0.2->Twisted==16.3.0)
Collecting wstools (from soappy->Twisted==16.3.0)
  Using cached wstools-0.4.4-py2.py3-none-any.whl
Collecting defusedxml (from soappy->Twisted==16.3.0)
  Using cached defusedxml-0.4.1.tar.gz
Collecting pytest-timeout (from wstools->soappy->Twisted==16.3.0)
  Using cached pytest_timeout-1.0.0-py2.py3-none-any.whl
Collecting six (from wstools->soappy->Twisted==16.3.0)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting pep8 (from wstools->soappy->Twisted==16.3.0)
  Using cached pep8-1.7.0-py2.py3-none-any.whl
Collecting pytest (from wstools->soappy->Twisted==16.3.0)
  Using cached pytest-2.9.2-py2.py3-none-any.whl
Collecting pytest-pep8 (from wstools->soappy->Twisted==16.3.0)
  Using cached pytest-pep8-1.0.6.tar.gz
Collecting autopep8 (from wstools->soappy->Twisted==16.3.0)
  Using cached autopep8-1.2.4-py2.py3-none-any.whl
Collecting pytest-cov (from wstools->soappy->Twisted==16.3.0)
  Using cached pytest_cov-2.3.0-py2.py3-none-any.whl
Collecting py>=1.4.29 (from pytest->wstools->soappy->Twisted==16.3.0)
  Using cached py-1.4.31-py2.py3-none-any.whl
Collecting pytest-cache (from pytest-pep8->wstools->soappy->Twisted==16.3.0)
  Using cached pytest-cache-1.0.tar.gz
Collecting coverage>=3.7.1 (from pytest-cov->wstools->soappy->Twisted==16.3.0)
  Using cached coverage-4.1-cp36-cp36m-macosx_10_10_x86_64.whl
Collecting execnet>=1.1.dev1 (from pytest-cache->pytest-pep8->wstools->soappy->Twisted==16.3.0)
  Using cached execnet-1.4.1-py2.py3-none-any.whl
Collecting apipkg>=1.4 (from execnet>=1.1.dev1->pytest-cache->pytest-pep8->wstools->soappy->Twisted==16.3.0)
  Using cached apipkg-1.4-py2.py3-none-any.whl
Installing collected packages: zope.interface, py, pytest, pytest-timeout, six, pep8, apipkg, execnet, pytest-cache, pytest-pep8, autopep8, coverage, pytest-cov, wstools, defusedxml, soappy, Twisted
  Running setup.py install for pytest-cache ... done
  Running setup.py install for pytest-pep8 ... done
  Running setup.py install for defusedxml ... done
  Running setup.py install for soappy ... done
  Running setup.py develop for Twisted
Successfully installed Twisted apipkg-1.4 autopep8-1.2.4 coverage-4.1 defusedxml-0.4.1 execnet-1.4.1 pep8-1.7.0 py-1.4.31 pytest-2.9.2 pytest-cache-1.0 pytest-cov-2.3.0 pytest-pep8-1.0.6 pytest-timeout-1.0.0 six-1.10.0 soappy-0.12.22 wstools-0.4.4 zope.interface-4.2.0
You are using pip version 8.1.1, however version 8.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

However, even with this change the SOAPpy library "pip installs" with Python 3, but it doesn't work on Python 3.
